### PR TITLE
test: add more positive fuzz test cases

### LINKED ISSUE #971: Add more positive fuzz tests to the test suite
Wonder if we should add some more fuzzing unit tests that show the success case, given here I think we have a lot of "failign" ones which we use to gain confidence

_Originally posted by @gakonst in https://github.com/gakonst/foundry/pull/965#discussion_r829496068_

### PR DESCRIPTION
**Pull request by malik672**
Closes #971

### CODE REVIEW

**Review by mattsse — APPROVED** [review]

**Review by mattsse — COMMENTED** [review]
I think this interferes with an existing test

**Comment by malik672** [comment]
> I think this interferes with an existing test

what do you think I should do

**Comment by malik672** [comment]
rewrite the tests?

**Comment by malik672** [comment]
i think i will run the ci on my computer, as I'm building

**Comment by malik672** [comment]
do you think the failure here might be to race conditons

**Review by mattsse — CHANGES_REQUESTED** [review]
the issue here is that these functions are being run in another test:

--- TRY 3 STDERR:        forge::it fuzz::test_fuzz ---
thread 'fuzz::test_fuzz' panicked at crates\forge\tests\it\fuzz.rs:37:22:
Test testSuccessChecker(uint256) did not fail as expected.
Reason: None
Logs:

best solution here would be to move the new tests to a new contract/file

**Comment by malik672** [comment]
ok

File: testdata/trace/Trace.t.sol [hunk 1]
```diff
@@ -7,7 +7,7 @@ contract RecursiveCall {
     TraceTest factory;
 
     event Depth(uint256 depth);
-    event ChildDepth(uint256 childDepth);
+    event ChildDepth(uint256 childDepth);
```
Comment (onbjerg) [inline testdata/trace/Trace.t.sol:10]:
```suggestion
    event ChildDepth(uint256 childDepth);
```

File: testdata/cheats/ReadCallers.t.sol [hunk 2]
```diff
@@ -14,7 +14,7 @@ contract ReadCallersTest is DSTest {
     function testReadCallersWithNoActivePrankOrBroadcast() public {
         address expectedSender = msg.sender;
         address expectedTxOrigin = tx.origin;
-
+
```
Comment (onbjerg) [inline testdata/cheats/ReadCallers.t.sol:17]:
```suggestion

```

File: testdata/cheats/Broadcast.t.sol [hunk 3]
```diff
@@ -94,7 +94,7 @@ contract BroadcastTest is DSTest {
     }
 
     function deployOther() public {
-        vm.startBroadcast(ACCOUNT_A);
+        vm.startBroadcast(ACCOUNT_A);
```
Comment (onbjerg) [inline testdata/cheats/Broadcast.t.sol:97]:
```suggestion
        vm.startBroadcast(ACCOUNT_A);
```

**Review by onbjerg — COMMENTED** [review]
small nits

**Comment by malik672** [comment]
I think I've fixed it

File: crates/forge/tests/it/fuzz.rs [hunk 4]
```diff
@@ -46,6 +46,39 @@ async fn test_fuzz() {
...
+            test_opts(),
+        )
+        .await;
+
+    assert!(!suite_result.is_empty());
+
+    for (_, SuiteResult { test_results, .. }) in suite_result {
+        for (test_name, result) in test_results {
+            match test_name.as_str() {
+                "testSuccessChecker1(uint256)" |
```
Comment (onbjerg) [inline crates/forge/tests/it/fuzz.rs:67]:
```suggestion
                "testSuccessChecker(uint256)" |
```

**Review by onbjerg — CHANGES_REQUESTED** [review]

**Review by malik672 — COMMENTED** [review]
done

**Review by onbjerg — APPROVED** [review]
thanks!

**Pull request merged**